### PR TITLE
Fix username checks; Simplify username and role checks

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/form/TagUtils.java [hunk 1]
```diff
@@ -15,4 +18,30 @@ public static boolean instanceOf(Object o, String className) {
     }
   }
 
+  public static boolean isCurrentUser(String username) {
+    return username.toLowerCase()
```
Comment (apmasell) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/form/TagUtils.java]: There's an `equalsIgnoreCase`.

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/form/TagUtils.java [hunk 2]
```diff
@@ -15,4 +18,30 @@ public static boolean instanceOf(Object o, String className) {
...
+    return hasRole("ROLE_ADMIN");
+  }
+
+  public static boolean isTech() {
+    return hasRole("ROLE_TECH");
+  }
+
+  private static boolean hasRole(String role) {
+    for (GrantedAuthority authority : getAuthentication().getAuthorities()) {
+      if (authority.getAuthority().toUpperCase().equals(role)) {
```
Comment (sqwci) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/form/TagUtils.java]: MINOR Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1157)

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/form/TagUtils.java (hunk 1 above)
Comment (sqwci) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/form/TagUtils.java]: MINOR Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1157)

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/form/TagUtils.java [hunk 3]
```diff
@@ -1,6 +1,9 @@
 package uk.ac.bbsrc.tgac.miso.webapp.util.form;
 
 import org.apache.log4j.Logger;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
 
 public class TagUtils {
```
Comment (sqwci) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/form/TagUtils.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)

**Comment by sqwci** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**